up42/http/session.py

```diff
@@ -0,0 +1,29 @@
+from functools import lru_cache
+
+import requests
+
+from up42.http import http_adapter, oauth
+
+SCHEMAS = ["http", "https"]
+
+
+class StatusValidatingSession(requests.Session):
+    def request(self, *args, **kwargs) -> requests.Response:
+        raise_for_status = kwargs.pop("raise_for_status", True)
+        response = super().request(*args, **kwargs)
+        if raise_for_status:
+            response.raise_for_status()
+        return response
+
+
+@lru_cache
+def create(
+    create_adapter=http_adapter.create,
+    create_auth=oauth.ProjectAuth,
+) -> requests.Session:
+    session = StatusValidatingSession()
+    adapter = create_adapter()
+    for schema in SCHEMAS:
+        session.mount(schema + "://", adapter)
```

Comment:
This part is not covered by tests, since it requires an integration test on network level, which is not supported by requests_mock